src/ffi/types.rs

```diff
@@ -9,23 +9,23 @@ use std::panic::RefUnwindSafe;
 use crate::device::{AdapterType, Error as DevError};
 
 pub trait TelioEventCb: Send + Sync + std::fmt::Debug {
-    fn event(&self, payload: String);
+    fn event(&self, payload: String) -> FFIResult<()>;
 }
 
 pub trait TelioLoggerCb: Send + Sync + std::fmt::Debug {
-    fn log(&self, log_level: TelioLogLevel, payload: String);
+    fn log(&self, log_level: TelioLogLevel, payload: String) -> FFIResult<()>;
 }
 
 pub trait TelioProtectCb: Send + Sync + RefUnwindSafe + std::fmt::Debug {
-    fn protect(&self, payload: i32);
+    fn protect(&self, payload: i32) -> FFIResult<()>;
 }
 
 pub type FFIResult<T> = Result<T, TelioError>;
 
 #[derive(Debug, thiserror::Error)]
 pub enum TelioError {
-    #[error("UnknownError: {0}")]
-    UnknownError(#[source] anyhow::Error),
+    #[error("UnknownError: {inner}")]
+    UnknownError { inner: String },
```

Comment:
Could you clarify why this change is needed?

Reply:
UniFFI doesn't like structs with unnamed fields, hence `{inner: String}` instead of `(String)`, but you probably mean why there is additional data for the error at all. In most cases it's whatever, but in the `catch_ffi_panic` function in `ffi/mod.rs` it makes it much easier to debug errors since we now keep the panic message

Reply:
I was also wondering why is that the previous version could use `anyhow::Error` and now we have downgraded it to an arbitrary `String`.

Reply:
That is because uniffi doesn't have a concept of what `anyhow::Error` is. I can either tell uniffi that it's a custom type and provide a mapping to and from `String` or declare `TelioError` as an interface of methods instead of an interface of variants. I think all three solutions (the two listed here and the currently implemented) have their downsides. I'll think about which one I prefer, but feel free to weigh in

Reply:
I see, I don't think the additional complexity is worth it - as far as I'm concerned feel free to keep the `String` here.